internal/pkg/bulk/engine.go

```diff
@@ -259,7 +260,9 @@ func (b *Bulker) Client() *elasticsearch.Client {
 }
 
 func (b *Bulker) RemoteOutputConfigChanged(zlog zerolog.Logger, name string, newCfg map[string]interface{}) bool {
+	b.remoteOutputMutex.RLock()
 	curCfg := b.remoteOutputConfigMap[name]
+	b.remoteOutputMutex.RUnlock()
```

Comment:
With this implementation, don't you run the risk of another goroutine change the value of `b.remoteOutputConfigMap[name]` after `curCfg` has been set, thereby rendering the value of `curCfg` stale?  Would the following implementation make more sense, as it would bring `curCfg` into the critical section protected by the mutex as well?

```suggestion
	b.remoteOutputMutex.RLock()
	defer b.remoteOutputMutex.RUnlock()
	curCfg := b.remoteOutputConfigMap[name]
```